### PR TITLE
Feature - Validate ui5.yaml file for adaptation projects

### PR DESCRIPTION
**Pull request by Dannysht**
feature for #1495 

- Added adaptation project properties validation for ui5 yaml files

### CODE REVIEW

File: packages/adp-tooling/src/base/validate.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * Assert ui5 for its validity. Throws error if a property of middlewares is missing.
+     *
+     * @param properties array of property name
+     * @param target object which will be checked
+     */
+    private static assertProperties<T extends object>(properties: Properties<T>[], target: T): void {
+        for (const property of properties) {
+            const value = target[property];
+            if (value === null || value === undefined) {
+                throw new Error(`Missing ${property} in the ui5 yaml file`);
```
Comment (longieirl) [inline packages/adp-tooling/src/base/validate.ts:42]: Would `throw new Error(`Missing ${property} in the ui5.yaml file`);` make more sense?

**Review by zdravko-georgiev — CHANGES_REQUESTED** [review]
This is not a good approach. This works only for  the oss middlewares. If you want to validate the configuration of adp, then you can do it e.g. in the constructor of the `ADPPreview`. Also it doesn't make any sense to validate the configuration of the other middlewares. Each middleware is exchangeable. You can have n-different middlewares that are proxying e.g. the UI5 resources.

File: packages/adp-tooling/src/base/validate.ts (hunk 1 above)
Comment (Dannysht) [reply packages/adp-tooling/src/base/validate.ts]: Changed in https://github.com/SAP/open-ux-tools/pull/1496/commits/6dc65595d09bb26940c13a19d9de92f9b87e56ce

File: packages/adp-tooling/src/base/validate.ts [hunk 2]
```diff
@@ -45,19 +47,25 @@ export default class UI5Validator {
...
         }
         this.assertProperties(['environment'], customConfiguration);
-        this.assertProperties(['configuration'], previewMiddleware);
-        this.assertProperties(['adp'], previewMiddleware.configuration);
-        this.assertProperties(['target'], previewMiddleware.configuration.adp);
-        this.assertProperties(['url', 'client'], previewMiddleware.configuration.adp.target);
+        this.assertProperties(['configuration'], fioriPreview);
+        this.assertProperties(['adp'], fioriPreview.configuration);
+        this.assertProperties(['target'], fioriPreview.configuration.adp);
+        this.assertProperties(['url', 'client'], fioriPreview.configuration.adp.target);
```
Comment (nikmace) [inline packages/adp-tooling/src/base/validate.ts:65]: When the user creates an ADP in BAS with the new structure, he will not have a url and client properties, which will throw an error and fail. The same will happen with the destination property on VS Code.

File: packages/adp-tooling/src/base/validate.ts [hunk 3]
```diff
@@ -45,19 +47,25 @@ export default class UI5Validator {
...
         customConfiguration: AdpCustomConfiguration | undefined,
-        previewMiddleware: CustomMiddleware<PreviewMiddlewareConfiguration> | undefined
+        fioriPreview: CustomMiddleware<FioriToolsPreviewConfig> | undefined,
+        fioriProxy: CustomMiddleware<FioriToolsProxyConfig> | undefined
     ): void {
-        if (!customConfiguration || !previewMiddleware) {
+        if (!customConfiguration || !fioriPreview || !fioriProxy) {
             throw new Error('Missing required custom middleware or custom configuration in ui5.yaml');
         }
         this.assertProperties(['environment'], customConfiguration);
```
Comment (nikmace) [inline packages/adp-tooling/src/base/validate.ts:61]: We have removed `environment` from customConfiguration. It is ABAP by default. We have added safeMode property to customConfiguration, this one probably needs to be checked.

**Review by nikmace — CHANGES_REQUESTED** [review]

File: packages/adp-tooling/src/base/validate.ts (hunk 2 above)
Comment (Dannysht) [reply packages/adp-tooling/src/base/validate.ts:65]: There is still no BAS implementation, and we are not populating destination in ui5.yaml if a project is generated from BAS

**Comment by zdravko-georgiev** [comment]
Let us please discuss in this in our sync meeting.

**Pull request closed without merging**